Update - praca dyplomowa (#18)
iiuni/poradnik: porady_odnosnie_studiowania.md

    Przepisywanie treści wykładu słowo w słowo nie jest najlepszą taktyką. Znacznie lepiej sprawdza się notowanie zwięzłe, tak, aby móc skupić się na zrozumieniu materiału, zamiast na idealnym skopiowaniu słów prowadzącego. Znajdź balans pomiędzy dokładnością notatek a skupieniem na wykładzie.
  </details>

---

## Praca dyplomowa

* **Wybierz interesujący Cię temat**

  <details>
    <summary>Szczegóły</summary>

>     Pracę dyplomową najlepiej pisać na temat na którym się znasz, który Cię interesuje i z którym wiążesz swoją przyszłość. Przede pomoże Ci to pracować nad nią systematycznie i przemóc ewentualną prokrastynację. Ponadto najpewniej zamieścisz tytuł i opis pracy (a może nawet repozytorium pod którym można znaleźć jej treść/kod napisany w jej ramach) w swoim CV/portfolio/liście intencyjnym/etc. - dlatego warto, żeby praca była najwyższej możliwej jakości w czym chęci do tworzenia jej zdecydowanie pomagają.

Reviewer: ```suggestion
    Pracę dyplomową najlepiej pisać na temat, na którym się znasz, który Cię interesuje i z którym wiążesz swoją przyszłość. Przede wszystkim pomoże Ci to pracować nad nią systematycznie i przemóc ewentualną prokrastynację. Ponadto najpewniej zamieścisz tytuł i opis pracy (a może nawet repozytorium, pod którym można znaleźć jej treść/kod napisany w jej ramach) w swoim CV/portfolio/liście intencyjnym/etc. - dlatego warto, żeby praca była najwyższej możliwej jakości, w czym chęci do tworzenia jej zdecydowanie pomagają.
```

  </details>

* **Pracuj systematycznie i zacznij wcześnie**

  <details>
    <summary>Szczegóły</summary>

>     Jakkolwiek kuszące może być odłożenie robienia pracy dyplomowej na ostatnią chwilę, jest to bardzo zła praktyka która źle odbije się na jej jakości, Twoim samopoczuciu i finalnej ocenie. Najlepszą strategią jest regularna pisanie pracy i jej komponentów, począwszy od początku semestru w który musisz ją oddać. Temat pracy i jej promotora warto wybrać jeszcze wcześniej.

Reviewer: ```suggestion
    Jakkolwiek kuszące może być odłożenie robienia pracy dyplomowej na ostatnią chwilę, jest to bardzo zła praktyka, która źle odbije się na jej jakości, Twoim samopoczuciu i finalnej ocenie. Najlepszą strategią jest regularna pisanie pracy i jej komponentów, począwszy od początku semestru, w który musisz ją oddać. Temat pracy i jej promotora warto wybrać jeszcze wcześniej.
```

  </details>

* **Promotor nie napisze za Ciebie pracy**

  <details>
    <summary>Szczegóły</summary>

>     Promotor w procesie pisania pracy dyplomowej jest przewodnikiem oraz oceniającym a nie jednym z jej autorów. Możesz zwrócić się do niego z pytaniami dotyczącymi np. technologii które chcesz wykorzystać czy propozycji rozwiązania/obejścia problemu na który się natknąłeś. Nie licz natomiast, że promotor będzie prowadził Cię za rękę lub pilnował czy pracujesz systematycznie.

Reviewer: ```suggestion
    Promotor w procesie pisania pracy dyplomowej jest przewodnikiem oraz oceniającym, a nie jednym z jej autorów. Możesz zwrócić się do niego z pytaniami dotyczącymi np. technologii, które chcesz wykorzystać czy propozycji rozwiązania/obejścia problemu, na który się natknąłeś. Nie licz natomiast, że promotor będzie prowadził Cię za rękę lub pilnował czy pracujesz systematycznie.
```

>     Ponadto powinieneś pamiętać, że Twój promotor najpewniej zajmuje się kilkoma pracami dyplomwymi naraz i każdej musi poświęcić uwagę.

Reviewer: ```suggestion
    Ponadto powinieneś pamiętać, że Twój promotor najpewniej zajmuje się kilkoma pracami dyplomowymi naraz i każdej musi poświęcić uwagę.
```

  </details>

* **Pamiętaj o terminach**

  <details>
    <summary>Szczegóły</summary>

>     Praca dyplomowa musi zostać wprowadzona do systemu APD (Archiwum Prac Dyplomowych) na minimum tydzień przed terminem prezentacji/obrony. Jednakże w systemie musi zostać zatwierdzona przez promotora a potem przepuszczona przez system antyplagiatowy (który potrafi sprawdzać pracę nawet do 2-3 dni), dlatego warto zamieścić ją w APD kilka dni przed obowiązkowym terminem.

Reviewer: ```suggestion
    Praca dyplomowa musi zostać wprowadzona do systemu APD (Archiwum Prac Dyplomowych) na minimum tydzień przed terminem prezentacji/obrony. Jednakże w systemie musi zostać zatwierdzona przez promotora, a potem przepuszczona przez system antyplagiatowy (który potrafi sprawdzać pracę nawet do 2-3 dni), dlatego warto zamieścić ją w APD kilka dni przed obowiązkowym terminem.
```

>     Praca dyplomowa musi zostać wprowadzona do systemu APD (Archiwum Prac Dyplomowych) na minimum tydzień przed terminem prezentacji/obrony. Jednakże w systemie musi zostać zatwierdzona przez promotora a potem przepuszczona przez system antyplagiatowy (który potrafi sprawdzać pracę nawet do 2-3 dni), dlatego warto zamieścić ją w APD kilka dni przed obowiązkowym terminem.

Reviewer: Czy nie ma już jakiegoś wymogu donoszenia pracy w wersji papierowej? Kiedyś to trzeba było robić.

Author: W tym roku zniesiono konieczność przedstawiania prac dyplomowych w wersji papierowej (na pewno lic. i inż., mgr. chyba też)

>     Ponadto pracę przed zamieszczeniem w APD czyta promotor i zwraca do niej zestaw uwag do których musisz się odnieść i poprawić wskazane w nich fragmenty. Warto zarezerwować na ten proces jeden do dwóch tygodni, zwłaszcza że praca może być odesłana do poprawki przez promotora kilkakrotnie.

Reviewer: ```suggestion
    Ponadto pracę przed zamieszczeniem w APD czyta promotor i zwraca do niej zestaw uwag, do których musisz się odnieść i poprawić wskazane w nich fragmenty. Warto zarezerwować na ten proces jeden do dwóch tygodni, zwłaszcza że praca może być odesłana do poprawki przez promotora kilkakrotnie.
```

  </details>

* **Przygotuj się do obrony/prezentacji**

  <details>
    <summary>Szczegóły</summary>

    Ocena pracy dyplomowej składa się z oceny samej pracy oraz oceny jej prezentacji/obrony przed promotorem oraz recenzentem. Dlatego ważne jest, aby dobrze przygotować się do zaprezentowania pracy.

>     Koniecznie przyjdź na obronę z własnym sprzętem - laptopem do którego można podłączyć standardowy wydziałowy rzutnik za pomocą kabla HDMI (kabel i rzutnik będą na Ciebie czekały).

Reviewer: ```suggestion
    Koniecznie przyjdź na obronę z własnym sprzętem - laptopem, do którego można podłączyć standardowy wydziałowy rzutnik za pomocą kabla HDMI (kabel i rzutnik będą na Ciebie czekały).
```

>     Prezentacja pracy składa się z dwóch części:

Reviewer: W przypadku pracy magisterskiej jest jeszcze egzamin magisterski ustny. Być może warto to jakoś zaznaczyć, albo dopisać że chodzi o prace licencjackie.

>     -   właściwej prezentacji podczas której omawiasz pracę (jej problematykę, wykorzystane technologie i rozwiązania etc.) - przygotuj tutaj prezentację multimedialną mającą 10-15 slajdów którą wygłosisz w 10-15 minut. Przećwicz ją przed lustrem/kaczuszką/współlokatorem aby mieć pewność, że potrafisz wygłośić ją płynnie i bez notatek. Prezentacja za krótka bądź za długa negatywnie odbije się na Twojej ocenie

Reviewer: ```suggestion
    -   właściwej prezentacji, podczas której omawiasz pracę (jej problematykę, wykorzystane technologie i rozwiązania etc.) - przygotuj tutaj prezentację multimedialną mającą 10-15 slajdów, którą wygłosisz w 10-15 minut. Przećwicz ją przed lustrem/kaczuszką/współlokatorem, aby mieć pewność, że potrafisz wygłosić ją płynnie i bez notatek. Prezentacja za krótka bądź za długa negatywnie odbije się na Twojej ocenie
```

    -   pytań od recenzenta i promotora - każdy z nich może zadać Ci do trzech pytań związanych z pracą dyplomową

>     Ponadto jeżeli w ramach swojej pracy zaimplementowałeś system/algorytm/coś czego działanie możesz pokazać podczas prezentacji pracy, będziesz miał na to do 10 minut między prezentacją a pytaniami od promotora i recenzenta. Upewnij się, że to co chcesz pokazać działa prawidłowo i dla pewności nagraj film (bez audio, komentuj co dzieje się na ekranie na żywo) na wypadek gdyby Twój laptop odmówił posłuszeństwa.

Reviewer: ```suggestion
    Ponadto, jeżeli w ramach swojej pracy zaimplementowałeś system/algorytm/coś, czego działanie możesz pokazać podczas prezentacji pracy, będziesz miał na to do 10 minut między prezentacją a pytaniami od promotora i recenzenta. Upewnij się, że to, co chcesz pokazać działa prawidłowo i dla pewności nagraj film (bez audio, komentuj co dzieje się na ekranie na żywo) na wypadek gdyby Twój laptop odmówił posłuszeństwa.
```

  </details>

* **Możesz odłożyć prezentację pracy**

  <details>
    <summary>Szczegóły</summary>

    Jeżeli nie uda Ci się napisać pracy tak, aby złożyć ją w wymaganym terminie, możesz odłożyć jej złożenie i prezentację na następny semestr. Być może będzie się to wiązało ze skreśleniem z listy studentów (patrz dalej), ale nie jest to koniec świata. O zamiarze odłożenia terminu złożenia pracy poinformuj promotora z wyprzedzeniem.

>     Ponadto jeżeli termin obrony Twojej pracy inżynierskiej/licencjackiej przypada na semestr letni, możesz prezentować ją na początku września, przed sesją poprawkową. W takim przypadku stracisz możliwość przystąpienia do egzaminu w lipcu i będziesz miał tylko jeden termin podczas sesji poprawkowej.

Reviewer: ```suggestion
    Ponadto, jeżeli termin obrony Twojej pracy inżynierskiej/licencjackiej przypada na semestr letni, możesz prezentować ją na początku września, przed sesją poprawkową. W takim przypadku stracisz możliwość przystąpienia do egzaminu w lipcu i będziesz miał tylko jeden termin podczas sesji poprawkowej.
```

  </details>

* **Niezłożenie pracy ma konsekwencje**

  <details>
    <summary>Szczegóły</summary>

>     Jeżeli zaliczyłeś wszystkie wymagania jakie należy zrealizować przed złożeniem pracy (obowiązki, efekty, ECTS etc.), zostaniesz skreślony z listy studentów i będziesz mógł złożyć i obronić pracę w terminie do dwóch lat (cztery semestry) od daty skreślenia. Aby móc złożyć pracę, będziesz musiał złożyć wniosek o przywrócenie na listę studentów (ze względu na chęć złożenia i obrony pracy dyplomowej) na miesiąc-dwa przed terminami ich prezentacji. Uwaga: po skreśleniu z listy studentów stracisz dostęp do niektórych zasobów uniwersyteckich (np. uczelniane konto Microsoft, co może utrudnić komunikację z promotorem).

Reviewer: ```suggestion
    Jeżeli zaliczyłeś wszystkie wymagania, jakie należy zrealizować przed złożeniem pracy (obowiązki, efekty, ECTS etc.), zostaniesz skreślony z listy studentów i będziesz mógł złożyć i obronić pracę w terminie do dwóch lat (cztery semestry) od daty skreślenia. Aby móc złożyć pracę, będziesz musiał złożyć wniosek o przywrócenie na listę studentów (ze względu na chęć złożenia i obrony pracy dyplomowej) na miesiąc-dwa przed terminami ich prezentacji. Uwaga: po skreśleniu z listy studentów stracisz dostęp do niektórych zasobów uniwersyteckich (np. uczelniane konto Microsoft, co może utrudnić komunikację z promotorem).
```

>     Jeżeli zaliczyłeś wszystkie wymagania jakie należy zrealizować przed złożeniem pracy (obowiązki, efekty, ECTS etc.), zostaniesz skreślony z listy studentów i będziesz mógł złożyć i obronić pracę w terminie do dwóch lat (cztery semestry) od daty skreślenia. Aby móc złożyć pracę, będziesz musiał złożyć wniosek o przywrócenie na listę studentów (ze względu na chęć złożenia i obrony pracy dyplomowej) na miesiąc-dwa przed terminami ich prezentacji. Uwaga: po skreśleniu z listy studentów stracisz dostęp do niektórych zasobów uniwersyteckich (np. uczelniane konto Microsoft, co może utrudnić komunikację z promotorem).

Reviewer: Czy termin złożenia wniosku o przywrócenie na listę studentów jest na miesiąc-dwa przed terminami prezentacji pracy? Rok temu kazali mi składać wniosek na kilka tygodni przed rozpoczęciem semestru w którym chciałem się bronić.

Author: Osobiście w tym roku (a w zasadzie to 2021/2022) otrzymałem maila z info o konieczności złożenia wniosku 27 maja i zaniosłem go w przeciągu bodajże dwóch tygodni

Reviewer: Tak, ale możliwe, że to się zmieniło, albo że miałem złe informacje. Zostawmy to póki co tak jak jest, wrzucę issue żeby to sprawdzić z dziekanatem.

    Jeżeli dalej masz do zrealizowania jakieś wymaganie/a (np. brakuje Ci kilku ectsów z jednej z grup), będziesz mógł kontynuować studia bez problemów.
  </details>

* **Możesz realizować pracę grupowo**

  <details>
    <summary>Szczegóły</summary>

    Jeżeli masz pomysł na pracę, której nie dasz radę sam zrealizować lub po prostu chciałbyś napisać pracę grupowo, możesz zrobić to w parze (lic/inż/mgr) lub w zespole trzyosobowym (tylko lic/inż). W takiej sytuacji stworzycie wspólnie jedną pracę, którą wspólnie zaprezentujecie podczas jednej, wspólnej obrony i otrzymacie za nią tę samą ocenę.

    Musisz pamiętać, że oprócz problemów związanych z samym pisaniem pracy, dojdą problemy związane z pisaniem jej w zespole - komunikacja, podział obowiązków etc. Rozwiązanie ich wymaga nieco uwagi i dobrze poświęcić ją na samym początku prac, aby później posuwały się gładko do przodu.

>     Każdy z członków zespołu realizującego wspólnie pracę dyplomową powinien włożyć w nią mniej więcej tyle samo wysiłku (doczepienie się do kogoś i napisanie pracy cudzymi rękami jest niedopuszczalne) w czym bardzo pomaga jasny i ustalony wspólnie podział obowiązków/odpowiedzialności. Dobrze jest gdy podczas spotkań z promotorem każdy z członków zespół referuje co udało mu się osiągnąć - z jednej strony pozwola to na kontrolowanie czy każdy realizuje swoją część a z drugiej podkreśla przed promotorem, że podział obowiązków jest sprawiedliwy i realizowany.

Reviewer: ```suggestion
    Każdy z członków zespołu realizującego wspólnie pracę dyplomową powinien włożyć w nią mniej więcej tyle samo wysiłku (doczepienie się do kogoś i napisanie pracy cudzymi rękami jest niedopuszczalne), w czym bardzo pomaga jasny i ustalony wspólnie podział obowiązków/odpowiedzialności. Dobrze jest, gdy podczas spotkań z promotorem każdy z członków zespołu referuje co udało mu się osiągnąć - z jednej strony pozwala to na kontrolowanie czy każdy realizuje swoją część, a z drugiej podkreśla przed promotorem, że podział obowiązków jest sprawiedliwy i realizowany.
```
